modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,262 @@
+= Reusing Types from JSON Schema
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+JSON schema is a standard providing a format for what JSON data is required for a given application and how to interact with it. Many existing data formats or types are expressed using this standard. In particular, Rest APIs described by a RAML file are most likely to have their types defined in JSON schemas.
+
+In order to reuse these already defined types, you can load the JSON schema into DataWeave leaving the types structure ready to be used in your scripts.
+
+The JSON schema loader parses schemas following the https://json-schema.org/specification-links.html#draft-7[JSON Schema Draft 7].
+
+== Import a Type from a JSON Schema
+
+Having a JSON schema in the resources: 
+
+.JSON Script
+[source,JSON]
+----
+{
+  "$id": "https://example.com/person.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Person",
+  "type": "object",
+  "properties": {
+    "firstName": {
+      "type": "string",
+      "description": "The person's first name."
+    },
+    "lastName": {
+      "type": "string",
+      "description": "The person's last name."
+    },
+    "age": {
+      "description": "Age in years which must be equal to or greater than zero.",
+      "type": "integer",
+      "minimum": 0
+    }
+  },
+  "required": ["firstName", "lastName"]
+}
+----
+
+The following example shows how you can use the JSON schema loader to import the type:
+
+.DataWeave file header
+----
+import * from jsonschema!example::schema::Person
+----
+
+== Import Syntax
+
+In order to import the JSON schema types, as shown above, use the following directive:
+
+----
+import _typesToImport_ from jsonschema!_pathToJsonSchema_
+----
+
+The following example shows how to import a type:
+
+----
+import * from jsonschema!example::schema::Person
+----
+
+Where:
+
+* `typeToImport`: You can replace it by `\*` to import all types defined in the schema or to export a single type from the schema, for example, *Root*. You can also import JSON schema types with a different name, for example, *Root as Person*. This way you can reference the type with that name in the script.
+* `pathToJsonSchema`: This represents the path to the schema file, modifying the file separators for `::` and removing the “.json” extension from the file name. For example, if the path to the schema is “example/schema/Person.json”, “example::schema::Person” must be used.
+
+== Use Your Types in a DataWeave Script
+
+Having this directive in the script header loads all the existing types in the JSON schema. In this case, the only existing type is the *Root* type, specified at the root which describes an Object that can have three properties (`firstName`, `lastName`, and `age`). It is equivalent to declaring the following type in your DataWeave script:
```

Comment:
"this directive" means which directive?

Reply:
I'm a bit confused by the paragraph. It's saying that the type definition in the DataWeave script describes and object with three fields. I'm just not clear what it's comparing that DataWeave type definition to.

Reply:
*Root* -> `Root` type
because I think it's referring to `type Root` in the script. The asterisks make that bold in the HTML and lack the code font.